backend/clubs/serializers.py

```diff
@@ -2222,8 +2228,120 @@ class Meta:
+    committees = ApplicationCommitteeSerializer(
+        many=True, required=False, read_only=True
+    )
+
+    class Meta:
+        model = ApplicationQuestion
+        fields = (
+            "id",
+            "question_type",
+            "prompt",
+            "word_limit",
+            "multiple_choice",
+            "committees",
+            "question_type",
+            "committee_question",
+            "precedence",
+        )
+
+    def create(self, validated_data):
+        # remove club from request we do not use it
+        validated_data.pop("club")
+
+        application_pk = self.context["view"].kwargs.get("application_pk")
+        validated_data["application"] = ClubApplication.objects.filter(
+            pk=application_pk
+        ).first()
+        return super().create(validated_data)
+
+    def save(self):
+        question_obj = super().save()
+        # manually create multiple choice answers as Django does not
+        # support nested serializers out of the box
+        request = self.context["request"].data
+        if "multiple_choice" in request:
+            multiple_choice = request["multiple_choice"]
+            ApplicationMultipleChoice.objects.filter(question=question_obj).delete()
+            for choice in multiple_choice:
+                ApplicationMultipleChoice.objects.create(
+                    value=choice["value"], question=question_obj,
+                )
+
+        # manually create committee choices as Django does not
+        # support nested serializers out of the box
+        if "committees" in request:
+            committees = request["committees"]
+            question_obj.committees.clear()
+            for committee in committees:
+                committee_obj = ApplicationCommittee.objects.filter(
+                    application=question_obj.application, name=committee["name"]
+                ).first()
+                question_obj.committees.add(committee_obj)
+
+        return question_obj
+
+
+class ApplicationQuestionResponseSerializer(serializers.ModelSerializer):
+    multiple_choice = ApplicationMultipleChoiceSerializer(
+        required=False, read_only=True
+    )
+    question_type = serializers.CharField(
+        source="question.question_type", read_only=True
+    )
+    question = ApplicationQuestionSerializer(required=False, read_only=True)
+
+    class Meta:
+        model = ApplicationQuestionResponse
+        fields = ("text", "multiple_choice", "question_type", "question")
+
+
+class ApplicationSubmissionSerializer(serializers.ModelSerializer):
+    committee = ApplicationCommitteeSerializer(required=False, read_only=True)
+    responses = ApplicationQuestionResponseSerializer(
+        many=True, required=False, read_only=True
+    )
+    user_hash = serializers.SerializerMethodField("get_user_hash")
+
+    def get_user_hash(self, obj):
+        username = obj.user.username
+        return int(hashlib.sha1(username.encode("utf-8")).hexdigest(), 16) % (10 ** 8)
```

Comment:
I assume this isn't supposed to be sensitive, but do you think it makes sense to hash the concatenation of PennKey and date joined or something like that just so that way it's much harder to reverse engineer values?